lib/matplotlib/axes/_axes.py

```diff
@@ -4559,6 +4559,16 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             collection.set_cmap(cmap)
             collection.set_norm(norm)
             collection._scale_norm(norm, vmin, vmax)
+        else:
+            extra_kwargs = {
+                    'cmap': cmap, 'norm': norm, 'vmin': vmin, 'vmax': vmax
+                    }
+            extra_keys = [k for k, v in extra_kwargs.items() if v is not None]
+            if any(extra_keys):
+                keys_str = ", ".join(f"'{k}'" for k in extra_keys)
+                _api.warn_external(
+                        "'c' is empty or not given, "
+                        f"kwargs {keys_str} will be ignored")
```

Comment:
```suggestion
                    "No data for colormapping provided via 'c'. "
                    f"Parameters {keys_str} will be ignored")
```
Prevents the "empty or not given" dance and relates 'c' to the "colormapping" sematics, which should make it more understandable why the parameters are ignored.

Also, don't use the term *kwargs*, which is jargon not necessarily understood by all users.

Reply:
🤦🏻 I have blindness to kwargs that I need to work on.